src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1630,6 +1633,112 @@ export class ResultsViewPageStore {
         }
     });
 
+    /**
+     * Fetch fusion data from the backend API with following request parameters:
+     * - entrez gene ids
+     * - sample molecular identifiers
+     * - molecular profile ids (only when sample molecular identifiers are not provided)
+     * @type {MobxPromiseUnionType<StructuralVariant[]>}
+     */
+    readonly fusions = remoteData<StructuralVariant[]>({
+        await: () => [
+            this.genes,
+            this.selectedMolecularProfiles,
+            this.samples,
+            this.studyIdToStudy
+        ],
+        invoke: async () => {
+            let molecularProfiles: Array<string> = [];
+            let samples: SampleMolecularIdentifier[] = [];
+            let params: StructuralVariantFilterExt = {entrezGeneIds:[]};
+            if (this.genes.result) {
+                params.entrezGeneIds = _.map(this.genes.result, (gene: Gene) => gene.entrezGeneId);
+            }
+            // compose request parameters
+            if (this.samples.result) {
+                // get sample molecular identifiers
+                samples = this.samples.result.map(sample => {
+                    return {
+                        molecularProfileId: sample.studyId + '_structural_variants',
+                        sampleId: sample.sampleId
+                    };
+                });
+                params.sampleMolecularIdentifiers = samples;
+            } else {
+                // get molecular profile ids
+                if (this.selectedMolecularProfiles.result) {
+                    molecularProfiles = this.selectedMolecularProfiles.result.filter(profile => {
+                        return profile.molecularAlterationType === AlterationTypeConstants.STRUCTURAL_VARIANT;
+                    }).map(profile => {
+                        return profile.molecularProfileId;
+                    });
+                    params.molecularProfileIds = molecularProfiles;
+                }
+            }
+            return await client.fetchStructuralVariantsUsingPOST({'structuralVariantFilter':<StructuralVariantFilter> params});
+        }
+    });
+
+    /**
+     * Group fusion data by genes.
+     * @returns {{[p: string]: StructuralVariant[]}}
+     */
+    @computed
+    get fusionsByGene(): { [hugeGeneSymbol: string]: StructuralVariant[] } {
```

Comment:
Often it makes most sense in a case like this, where the result depends on a promise being ready, that you turn this into a mobxpromise

```
readonly fusionsByGene = remoteData({
    await:()=>[this.genes],
   ...
```

Reply:
based on use below it seems this would be the best way to do it